crates/kas-core/src/event/cx/press.rs

```diff
@@ -108,23 +109,24 @@ impl PressSource {
         self.0 & Self::FLAG_TOUCH != 0
     }
 
-    /// Returns the touch identifier if this represents a touch event
-    fn touch_id(self) -> Option<u64> {
+    /// Returns the finger identifier if this represents a touch event
+    fn finger_id(self) -> Option<FingerId> {
         if self.is_touch() {
-            Some(self.0 & !Self::FLAG_TOUCH)
+            let id = self.0 & !Self::FLAG_TOUCH;
+            Some(FingerId::from_raw(id.cast()))
         } else {
             None
         }
     }
 
     /// Identify the mouse button used
     ///
-    /// This always returns `Some(_)` for mouse events and `None` for touch
-    /// events.
+    /// This returns `Some(button)` for mouse events with a button. It returns
+    /// `None` for touch events and mouse events without a button (e.g. motion).
     pub fn mouse_button(self) -> Option<MouseButton> {
```

Comment:
`PressSource::mouse_button` is now optional in that a mouse source may not have a button. Previously we reported a fake button for mouse motion without click (only reported to popup parents).